webapp/services.py

```diff
@@ -171,20 +196,92 @@ def create_tags_if_not_exist(self, tag_names):
         db_session.commit()
         return [*existing_tags, *tags_to_create]
 
+    def create_products_if_not_exists(self, product_names):
+        """
+        Create the product objects and return the
+        object from the database
+        """
+        if product_names == [""]:
+            return []
+
+        product_objects = []
+        for product_name in product_names:
+            existing_product = (
+                db_session.query(Product).filter_by(name=product_name).first()
+            )
+
+            if existing_product:
+                product_objects.append(existing_product)
+            else:
+                product = Product(name=product_name, assets=[])
+                product_objects.append(product)
+                db_session.add(product)
+
+        db_session.commit()
+        return product_objects
+
+    def create_author_if_not_exist(self, author):
+        """
+        Create the author object and return the object from the database
+        """
+        if not author:
+            return None
+
+        existing_author = (
+            db_session.query(Author).filter_by(email=author["email"]).first()
+        )
+
+        if existing_author:
+            return existing_author
+
+        author = Author(
+            first_name=author["first_name"],
+            last_name=author["last_name"],
+            email=author["email"],
+        )
+        db_session.add(author)
+        db_session.commit()
+        return
+
     def normalize_tag_name(self, tag_name):
         return tag_name.strip().lower()
 
-    def update_asset(self, file_path, tags=[], deprecated=None):
+    def update_asset(
+        self,
+        file_path,
+        tags=[],
+        deprecated=None,
+        products=[],
+        asset_type=None,
+        author=None,
+        google_drive_link=None,
+        salesforce_campaign_id=None,
+        language=None,
+    ):
```

Comment:
Since we are already providing the default values when we call the function, we might use type hinting here and that would seem more helpful